src/configure/configurers/provisioningConfigurer.ts

```diff
@@ -0,0 +1,280 @@
+export class ProvisioningConfigurer implements IProvisioningConfigurer {
+    private provisioningServiceClient: IProvisioningServiceClient;
+    private queuedPipelineUrl: string;
+    private refreshTime: number = 5 * 1000;
+    private maxNonStatusRetry: number = 60; // retries for max 5 min
+    private localGitRepoHelper: LocalGitRepoHelper;
+    private filesToCommit: DraftFile[] = [];
+
+    constructor(localGitRepoHelper: LocalGitRepoHelper){
+        this.localGitRepoHelper = localGitRepoHelper;
+    }
+
+    public async queueProvisioningPipelineJob(provisioningConfiguration: ProvisioningConfiguration, wizardInputs: WizardInputs): Promise<ProvisioningConfiguration>{
+        try {
+            this.provisioningServiceClient =  await ProvisioningServiceClientFactory.getClient(wizardInputs.githubPATToken, wizardInputs.azureSession.credentials);
+            const OrgAndRepoDetails = wizardInputs.sourceRepository.repositoryId.split('/');
+            return await this.provisioningServiceClient.createProvisioningConfiguration(provisioningConfiguration, OrgAndRepoDetails[0], OrgAndRepoDetails[1]);
+        } catch (error){
+            telemetryHelper.logError(Layer, TracePoints.UnableToCreateProvisioningPipeline, error);
+            throw error;
+        }
+    }
+
+    public async getProvisioningPipeline(jobId: string, githubOrg: string, repository: string, wizardInputs: WizardInputs): Promise<ProvisioningConfiguration>{
+       try {
+        this.provisioningServiceClient =  await ProvisioningServiceClientFactory.getClient(wizardInputs.githubPATToken, wizardInputs.azureSession.credentials);
+        return await this.provisioningServiceClient.getProvisioningConfiguration(jobId, githubOrg, repository);
+       } catch (error){
+            telemetryHelper.logError(Layer, TracePoints.UnabletoGetProvisioningPipeline, error);
+            throw error;
+       }
+    }
+
+    public async awaitProvisioningPipelineJob(jobId: string, githubOrg: string, repository: string, wizardInputs: WizardInputs): Promise<ProvisioningConfiguration> {
+        let statusNotFound: number = 0;
+        try {
+            const provisioningServiceResponse = await this.getProvisioningPipeline(jobId, githubOrg, repository, wizardInputs);
+            if ( provisioningServiceResponse.result){
+                if ( provisioningServiceResponse.result.status ===  "Queued" ||  provisioningServiceResponse.result.status == "InProgress") {
+                await sleepForMilliSeconds(this.refreshTime);
+                return await this.awaitProvisioningPipelineJob(jobId, githubOrg, repository, wizardInputs);
+                } else if (provisioningServiceResponse.result.status ===  "Failed") {
+                throw new Error(provisioningServiceResponse.result.message) ;
+                } else {
+                    return provisioningServiceResponse;
+                }
+            }else {
+                if ( statusNotFound < this.maxNonStatusRetry ) {
```

Comment:
so, this will get executed when `provisioningServiceResponse` is present but `provisioningServiceResponse.Result` is null. What case is that?
Should we re-try if any sort of failure happens in `getProvisioningPipeline`?

Reply:
Yes, this can happen if at provisioning service before sending response we are not properly converting to contracts or result  (generally i don't think this will happen, as at provisioning service we have proper conversion).
For re-try in `getProvisioningPipeline`, this function is quite simple and for failure in http request we have retry logic in ms-rest client.